src/main/java/rife/engine/TomcatServer.java

```diff
@@ -72,17 +74,33 @@ public TomcatServer addUser(String user, String pass) {
     }
 
     /**
-     * Add the location of the webapp directory or WAR for the root context.
+     * Adds a web application to the webapps directory.
      *
-     * @param docBase the base directory for the context, for static file.
+     * @param docBase   the base directory for the context, for static file
+     * @param isContext {@code true} to add as a context with no default {@code web.xml}, {@code false} otherwise
      * @return the instance of the server that's being configured
-     * @since 1.7.1
+     * @see #addWebapp(String)
+     * @since 1.8.1
      */
-    public TomcatServer addWebapp(String docBase) {
+    public TomcatServer addWebapp(String docBase, boolean isContext) {
```

Comment:
Instead of using an `isContext` flag, this could probably just be name `addContext` like the tomcat API.

Reply:
That's the way I first implemented it, but in Tomcat you can use both addContext and addWebapp at the same time, which is not what we want.

In this case, addWebapp means add the rife2 webapp to Tomcat.

I'm open to doing another way if you have a better idea.

Reply:
Can't this be like this?
```
public TomcatServer addContext(String docBase) {
    isContext_ = true;
    docBase_ = docBase;
    return this;
}
```

Reply:
That's exactly the way I had it, but that's very confusing, if you look at the underlying API.

I have an idea. Updated PR coming up.

Reply:
After much debating with my evil self. I decided to go the way I had first implemented it as `addContext`, even though it sets up a bunch of defaults contrary to the TC Internal API intent.
 
Feel free to squash the commits.